src/main/java/cuchaz/enigma/translation/mapping/serde/TinyMappingsWriter.java

```diff
@@ -0,0 +1,144 @@
+
+        Translator translator = new MappingTranslator(mappings, VoidEntryResolver.INSTANCE);
+
+        EntryMapping mapping = mappings.get(entry);
+        if (mapping != null && !entry.getName().equals(mapping.getTargetName())) {
+            if (entry instanceof ClassEntry) {
+                writeClass(writer, (ClassEntry) entry, translator);
+            } else if (entry instanceof FieldEntry) {
+                writeLine(writer, serializeEntry(entry, mapping.getTargetName()));
+            } else if (entry instanceof MethodEntry) {
+                writeLine(writer, serializeEntry(entry, mapping.getTargetName()));
+            }
+        }
+
+        writeChildren(writer, mappings, node);
+    }
+
+    private void writeChildren(Writer writer, EntryTree<EntryMapping> mappings, EntryTreeNode<EntryMapping> node) {
+        node.getChildren().stream()
+                .filter(e -> e instanceof FieldEntry).sorted()
+                .forEach(child -> writeEntry(writer, mappings, child));
+
+        node.getChildren().stream()
+                .filter(e -> e instanceof MethodEntry).sorted()
+                .forEach(child -> writeEntry(writer, mappings, child));
+
+        node.getChildren().stream()
+                .filter(e -> e instanceof ClassEntry).sorted()
+                .forEach(child -> writeEntry(writer, mappings, child));
+    }
+
+    private void writeClass(Writer writer, ClassEntry entry, Translator translator) {
+        ClassEntry translatedEntry = translator.translate(entry);
+
+        String obfClassName = entry.getFullName();
+        String deobfClassName = translatedEntry.getFullName();
+        writeLine(writer, new String[]{"CLASS", obfClassName, deobfClassName});
+    }
+
+    private void writeLine(Writer writer, String[] data) {
+        try {
+            String line = TAB_JOINER.join(data) + "\n";
+            if (writtenLines.add(line)) {
+                writer.write(line);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String[] serializeEntry(Entry<?> entry, String... extraFields) {
+        String[] data = null;
+
+        if (entry instanceof FieldEntry) {
+            data = new String[4 + extraFields.length];
+            data[0] = "FIELD";
+            data[1] = entry.getContainingClass().getFullName();
+            data[2] = ((FieldEntry) entry).getDesc().toString();
+            data[3] = entry.getName();
+        } else if (entry instanceof MethodEntry) {
+            data = new String[4 + extraFields.length];
+            data[0] = "METHOD";
+            data[1] = entry.getContainingClass().getFullName();
+            data[2] = ((MethodEntry) entry).getDesc().toString();
+            data[3] = entry.getName();
+        } else if (entry instanceof ClassEntry) {
+            data = new String[2 + extraFields.length];
+            data[0] = "CLASS";
+            data[1] = ((ClassEntry) entry).getFullName();
+        }
```

Comment:
Consider serializing parameters (which has `MTH-ARG` identifier, format can be found in tiny (v1) reader)

Reply:
This class is copied from weave. The current `tinyfy` task doesn't add parameters, and modmuss said he'd like to keep the yarn outputs the same for now.